spec/integration/puppet_x/puppetlabs/scheduled_task/taskscheduler2_task_spec.rb

```diff
@@ -0,0 +1,190 @@
+
+#!/usr/bin/env ruby
+require 'spec_helper'
+
+require 'puppet_x/puppetlabs/scheduled_task/taskscheduler2_task' if Puppet.features.microsoft_windows?
+
+RSpec::Matchers.define :be_same_as_powershell_command do |ps_cmd|
+  define_method :run_ps do |cmd|
+    full_cmd = "powershell.exe -NoLogo -NoProfile -NonInteractive -Command \"#{cmd}\""
+
+    result = `#{full_cmd}`
+
+    result.strip
+  end
+
+  match do |actual|
+    from_ps = run_ps(ps_cmd)
+
+    # This matcher probably won't tolerate UTF8 characters
```

Comment:
You can remove that comment... `==` should work fine for Unicode(*).


(*) There are some cases where Ruby can have problems - but its typically only when comparing normalized to non-normalized Unicode, which you wouldn't have to worry about here. See http://blog.honeybadger.io/ruby-s-unicode-support/

Reply:
Also, if you're concerned about the output coming back from PS itself ... multiple responses would come back as the same sequence of bytes (if corrupted by capturing things over `stdout`), so should also match for that reason.

Reply:
Okay, I pulled this comment over when I stole the method itself from Glenn's work on the adapter integration tests.